ui/app/AppLayouts/Wallet/panels/ManageHiddenPanel.qml

```diff
@@ -0,0 +1,227 @@
+
+    required property var assetsController
+    required property var collectiblesController
+
+    readonly property bool dirty: root.assetsController.dirty || root.collectiblesController.dirty
+    readonly property bool hasSettings: root.assetsController.hasSettings || root.collectiblesController.hasSettings
+
+    background: null
+
+    function clearSettings() {
+        root.assetsController.clearSettings();
+        root.collectiblesController.clearSettings();
+    }
+
+    QtObject {
+        id: d
+
+        property bool assetsExpanded: true
+        property bool collectiblesExpanded: true
+
+        readonly property int assetsCount: root.assetsController.hiddenTokensModel.count + root.assetsController.hiddenCommunityTokenGroupsModel.count
+        readonly property int collectiblesCount: root.collectiblesController.hiddenTokensModel.count + root.collectiblesController.hiddenCommunityTokenGroupsModel.count
+
+        readonly property var filteredHiddenAssets: SortFilterProxyModel {
+            sourceModel: root.assetsController.hiddenTokensModel
+            filters: FastExpressionFilter {
+                expression: {
+                    root.assetsController.hiddenCommunityGroups
+                    return !root.assetsController.hiddenCommunityGroups.includes(model.communityId)
+                }
+                expectedRoles: ["communityId"]
+            }
+        }
+
+        readonly property var filteredHiddenCollectibles: SortFilterProxyModel {
+            sourceModel: root.collectiblesController.hiddenTokensModel
+            filters: FastExpressionFilter {
+                expression: {
+                    root.collectiblesController.hiddenCommunityGroups
+                    return !root.collectiblesController.hiddenCommunityGroups.includes(model.communityId)
+                }
+                expectedRoles: ["communityId"]
+            }
+        }
+
+        readonly property var combinedModel: ConcatModel {
+            sources: [
+                SourceModel { // single hidden assets (not belonging to a group)
+                    model: d.filteredHiddenAssets
+                    markerRoleValue: "asset"
+                },
+                SourceModel { // community asset groups
+                    model: root.assetsController.hiddenCommunityTokenGroupsModel
+                    markerRoleValue: "assetGroup"
+                },
+                SourceModel { // single hidden collectibles (not belonging to a group)
+                    model: d.filteredHiddenCollectibles
+                    markerRoleValue: "collectible"
+                },
+                SourceModel { // community collectible groups
+                    model: root.collectiblesController.hiddenCommunityTokenGroupsModel
+                    markerRoleValue: "collectibleGroup"
+                }
+            ]
+
+            markerRoleName: "tokenType"
+        }
+
+        readonly property var sfpm: SortFilterProxyModel {
```

Comment:
Instead of adding SFPM to add `isCollectible` and `isGroup` we could use `tokenType` directly in delegates. It would be more efficient than additional proxy on top.

Reply:
We (will) need that SFPM anyway for sorting (most recent first), so I just took the opportunity to reuse it for this purpose too

Reply:
And no, we can't do it that way; either I use the `ConcatModel` above (with 4 source models, which would create 4 sections); or I use the same tag but then I can't differentiate between single/grouped items